plugin/pkg/shoot/validator/admission.go

```diff
@@ -134,20 +135,60 @@ func (v *ValidateShoot) Admit(a admission.Attributes, o admission.ObjectInterfac
 		return nil
 	}
 
-	// Ignore updates to shoot status or other subresources
-	if a.GetSubresource() != "" {
```

Comment:
I see why you removed this, but I guess it's still better to exit early. If the subresource is not empty then we already know that we definitely don't want to do anything in this admission plugin.